From: Ben Hutchings <ben@decadent.org.uk>
Date: Thu, 9 Sep 2010 03:09:03 +0100
Subject: [PATCH] sched: Avoid ABI change from introduction of thread_group_times

Move the new fields in struct signal_struct to the end and hide them
from genksyms.
---
 include/linux/sched.h |    9 ++++++---
 1 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/include/linux/sched.h b/include/linux/sched.h
index 3f77ccf..4b3dbc7 100644
--- a/include/linux/sched.h
+++ b/include/linux/sched.h
@@ -628,9 +628,6 @@ struct signal_struct {
 	cputime_t utime, stime, cutime, cstime;
 	cputime_t gtime;
 	cputime_t cgtime;
-#ifndef CONFIG_VIRT_CPU_ACCOUNTING
-	cputime_t prev_utime, prev_stime;
-#endif
 	unsigned long nvcsw, nivcsw, cnvcsw, cnivcsw;
 	unsigned long min_flt, maj_flt, cmin_flt, cmaj_flt;
 	unsigned long inblock, oublock, cinblock, coublock;
@@ -668,6 +665,12 @@ struct signal_struct {
 #endif
 
 	int oom_adj;	/* OOM kill score adjustment (bit shift) */
+
+#ifndef __GENKSYMS__
+#ifndef CONFIG_VIRT_CPU_ACCOUNTING
+	cputime_t prev_utime, prev_stime;
+#endif
+#endif
 };
 
 /* Context switch must be unlocked if interrupts are to be enabled */
-- 
1.7.1

